### ISSUE TITLE
how does it work with react-router?

### ISSUE THREAD

**Issue by lynnchin** [wmira/react-sidenav#71]
Hello, I am trying to use your library and I am confused how the react-router works with this library here. By looking at your example code (App.js), it seems that you are using MemoryRouter as Router. 

However, I am using something else (at least not MemoryRouter!) but I couldn't get it working by picking up the routing path and navigate to the page I want to go to.

```
import React from "react";

import { AppNavigation } from "./AppNavigation";
import { MemoryRouter as Router, Route, Switch } from "react-router-dom";

import { AppContainer, Navigation, Body, Title } from "./containers";

import { Home } from "./Home";
import { Basic } from "./Basic";
import { RenderItems } from "./RenderItems";
import { RenderItems2 } from "./RenderItems2";

class App extends React.Component {
  render() {
    return (
      <AppContainer>
        <Navigation>
          <Title> React SideNav </Title>
          <AppNavigation />
        </Navigation>
        <Body>
          <Switch>
            <Route path="//basic" component={Basic} />
            <Route path="//renderitems" component={RenderItems} />
            <Route path="//renderitems2" component={RenderItems2} />
            <Route path="/" exact component={Home} />
          </Switch>
        </Body>
      </AppContainer>
    );
  }
}

export const createApp = () => {
  return class SideNavApp extends React.Component {
    render() {
      return (
        <Router>
          <App />
        </Router>
      );
    }
  };
};
```

I have tried to use the id to match with the routing path but it is not picking up correctly.

```
<SideNav defaultSelectedPath={"home"}>
        <Nav id="home">
          <NavIcon>
            <Icon icon={home} />
          </NavIcon>
          <Text>Home</Text>
        </Nav>
```

So, I would like to know how does the id links to the react router path? See above, the id is "home", as the path is specified as "/"??? I am confused.

**Comment by wmira** [comment]
Hi @lynnchin have you checked this example?: https://github.com/wmira/react-sidenav/blob/master/src/playground/rr4.tsx - this one uses the BrowserRouter

It will try to get the path from the current active route e.g. /something and then if that matches then it will make it active

**Comment by lynnchin** [comment]
I will give it a go and see how it goes @wmira! Thanks!

**Comment by fjhixiuxiu** [comment]
@wmira I ran playground and it's making sense. The only thing is is I directly go to the url with `localhost:8080/home`, it says `Cannot GET /home`. But if you click it from the nav, it works fine. Am I doing something wrong here?

**Comment by wmira** [comment]
@lynnchin @fjhixiuxiu  there is some issue with the playground for rrouter. You need to have historyApiFallback set to true -- i have updated this.
There was a bug with the default selected path. I have fixed this on 3.1.8

you can try this now npm run playground -- src/playground/rr4.tsx

make sure you install deps .

**Comment by wmira** [comment]
this should be fix on 3.1.9 as there was a gitlab pipeline jest build issue on node v11.
